### PR TITLE
Use `pauli_word_prefactor` in `stoch_pulse_grad`

### PR DESCRIPTION
**Pull request by dwierichs**
**Context:**
~~`stoch_pulse_grad` relies on `qml.pauli.is_pauli_word` to verify that the standard parameter-shift rule applies to a generating term in a pulse Hamiltonian. However, `is_pauli_word` allows for prefactors of the Pauli word, so that the returned derivative needs to be rescaled accordingly.~~

#4164 added `qml.pauli.pauli_word_prefactor`, which extracts the prefactor of a (rescaled) Pauli word, as identified by `qml.pauli.is_pauli_word`.

**Description of the Change:**
~~Extract the prefactor from the Pauli word-like object by converting to a `PauliSentence` and multiply the computed derivative with the prefactor.~~

This PR makes use of this new function.
Note that due to unexpected ordering of PR merging, this PR is no longer a bug fix but mostly cleans up by using `pauli_word_prefactor` instead of extracting the prefactor manually.

In addition, this PR silences a warning raised by `qml.eigvals` if the pulse generator has fewer than 5 wires.

**Benefits:**
~~Bugfix.~~

Less code duplication.
Remove unnecessary warning that will occur a lot for small-scale pulses.

**Possible Drawbacks:**
N/A

**Related GitHub Issues:**

### CODE REVIEW

File: tests/gradients/core/test_pulse_gradient.py [hunk 1]
```diff
@@ -343,8 +343,33 @@ def test_constant_rx(self, num_split_times, t):
         res = fn(qml.execute(tapes, dev, None))
         assert qml.math.isclose(res, -2 * jnp.sin(2 * p) * delta_t)
 
+    def test_constant_ry_rescaled(self):
```
Comment (dwierichs) [inline tests/gradients/core/test_pulse_gradient.py:346]: This test fails under master.

File: pennylane/gradients/pulse_gradient.py [hunk 2]
```diff
@@ -571,7 +572,7 @@ def _generate_tapes_and_cjacs(tape, idx, key, num_split_times, use_broadcasting)
             split_evolve_ops = _split_evol_ops(op, word, ob.wires, tau)
             tapes.extend(_split_evol_tapes(tape, split_evolve_ops, op_idx))
     avg_prefactor = (t1 - t0) / num_split_times
-    return cjacs, tapes, avg_prefactor
+    return cjacs, tapes, avg_prefactor * prefactor
```
Comment (dwierichs) [inline pennylane/gradients/pulse_gradient.py:575]: The `avg_prefactor` is used to rescale the Monte Carlo integration. Here we just hijack this factor and put the rescaling factor of the generating term on top.

**Review by Qottmann — APPROVED** [review]
Great catch and thanks for the quick fix @dwierichs 🚀

File: pennylane/gradients/pulse_gradient.py [hunk 3]
```diff
@@ -90,12 +91,17 @@ def _split_evol_ops(op, ob, tau):
         after_t = jax.numpy.array([tau, t1])
 
     if qml.pauli.is_pauli_word(ob):
-        prefactor = next(iter(qml.pauli.pauli_sentence(ob).values()))
+        prefactor = qml.pauli.pauli_word_prefactor(ob)
```
Comment (Qottmann) [inline pennylane/gradients/pulse_gradient.py:94]: nice!

File: pennylane/gradients/pulse_gradient.py [hunk 4]
```diff
@@ -90,12 +91,17 @@ def _split_evol_ops(op, ob, tau):
...
         word = qml.pauli.pauli_word_to_string(ob)
         insert_ops = [qml.PauliRot(shift, word, ob.wires) for shift in [np.pi / 2, -np.pi / 2]]
         coeffs = [prefactor, -prefactor]
     else:
-        eigvals = qml.eigvals(ob)
+        with warnings.catch_warnings():
+            if len(ob.wires) <= 4:
+                warnings.filterwarnings(
+                    "ignore", ".*the eigenvalues will be computed numerically.*"
+                )
```
Comment (Qottmann) [inline pennylane/gradients/pulse_gradient.py:103]: 👌

**Review by Qottmann — APPROVED** [review]
Great fix @dwierichs, thanks! 🚀

**Review by Jaybsoni — APPROVED** [review]
💯
